src/huggingface_hub/constants.py

```diff
@@ -68,8 +74,6 @@
 
 HUGGINGFACE_HUB_CACHE = os.getenv("HUGGINGFACE_HUB_CACHE", default_cache_path)
 
-HF_HUB_OFFLINE = os.environ.get("HF_HUB_OFFLINE", "AUTO").upper()
-if HF_HUB_OFFLINE in ENV_VARS_TRUE_VALUES:
-    HF_HUB_OFFLINE = True
-else:
-    HF_HUB_OFFLINE = False
```

Comment:
I suspect that `"AUTO"` never really worked as expected here. I refactored it to make it more readable keeping exactly the same functionality but if we want `"AUTO"` to be considered as True, we need to use  `ENV_VARS_TRUE_AND_AUTO_VALUES` (so `_is_true_or_auto()`).